pkg/osquery/tables/firefox_preferences/table.go

```diff
@@ -0,0 +1,84 @@
+package firefox_preferences
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go"
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type Table struct {
+	name	 string
+	logger log.Logger
+}
+
+const tableName = "kolide_firefox_preferences"
+
+func TablePlugin(_ *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("path"),
+	)
+
+	t := &Table{
+		name:		tableName,
+		logger: logger,
+	}
+
+	return table.NewPlugin(t.name, columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	return generateData(queryContext, t.logger)
+}
+
+func generateData(queryContext table.QueryContext, logger log.Logger) ([]map[string]string, error)  {
+	paths := tablehelpers.GetConstraints(queryContext, "path")
+
+	if len(paths) != 1 {
+		return nil, errors.Errorf("The %s table requires that you specify a constraint for path", tableName)
+	}
+
+	file, err := os.Open(paths[0])
+
+	if err != nil {
+		// TODO: Investigate what error message looks like. Add filepath possibly
+		return nil, err
+	}
```

Comment:
nit ... the usually formatting I've seen in go does the `if err != nil` on the line immediately after the err assignment

```suggestion
	file, err := os.Open(paths[0])
	if err != nil {
		// TODO: Investigate what error message looks like. Add filepath possibly
		return nil, err
	}
```